provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/README.md

```diff
@@ -0,0 +1,26 @@
+### Motivation
+
+We wanted to validate the Cross Data Center functionality of Keycloak in an actual cross data center setup for Keycloak application. As part of that we created a test suite that is currently hosted in the keycloak-benchmark and in future would move to a more appropriate place.
+
+### CrossDC Test Framework
+
+The current framework is made up of the below components
+
+- **Testsuite root directory**: keycloak-benchmark/provision/rosa-cross-dc
+- **Test Runner**: JUnit5.
+- **Test Data**: Use [Keycloak Dataset provider](https://www.keycloak.org/keycloak-benchmark/dataset-guide/latest/).
+- **Cache Metrics**: [ISPN Http REST client](https://infinispan.org/docs/stable/titles/rest/rest.html) to access Cache stats for external ISPN server. And for embedded infinispan cache we are relying on [Dataset provider](https://www.keycloak.org/keycloak-benchmark/dataset-guide/latest/).
+- **Execution Target**: A CrossDC cluster with access to two Keycloak and Infinispan datacenter urls.
+
+<br/> Note: We will use the existing ROSA OCP cluster based deployment during development to bring up the cross-dc cluster.
+
+### How to run
+
+From the Testsuite root directory run the below command to run the tests
+
+```
+mvn clean install -DcrossDCTests \
```

Comment:
Yeah, I created the README.md before I added this script, makes sense to have it in the README.md, will add it in my next commit.

Reply:
seems also redundant when I wrote it down, what do you think about this below text to be added to README ? @mhajas 

-------

Alternatively could use the `run-crossdc-tests.sh` (located in the Testsuite root) directory to execute the tests when using a ROSA style provisioning setup to fetch the `ISPN_PASSWORD` on the fly, or by setting it manually.

Example usage:
```
./run-crossdc-tests.sh ISPN_DC1_URL=<ISPN_DC1_URL> ISPN_DC2_URL=<ISPN_DC2_URL> \
KEYCLOAK_DC1_URL=<KEYCLOAK_DC1_URL> KEYCLOAK_DC2_URL=<KEYCLOAK_DC2_URL>
```

Reply:
Having the script in the parent directory makes sense if we want to keep the Maven module environment agnostic. The other way, that we are using for some other scripts, is to define env variables before running the script like this:
```
ISPN_DC1_URL=<ISPN_DC1_URL> ISPN_DC2_URL=<ISPN_DC2_URL> KEYCLOAK_DC1_URL=<KEYCLOAK_DC1_URL> KEYCLOAK_DC2_URL=<KEYCLOAK_DC2_URL> ./run-crossdc-tests.sh
```

I don't mind any of the ways, although if we want to have the Maven module environment agnostic we may want to move it to the top directory instead of `provision/rosa-cross-dc`. Personally, I would prefer having it where it is now and see where this goes, maybe this will be moved to the `keycloak` repository eventually, who knows.